plugins/modules/file_remove.py

```diff
@@ -0,0 +1,290 @@
+  - By default, only files in the specified directory are removed (non-recursive).
+    Use the O(recursive) option to search and remove files in subdirectories.
+
+version_added: "12.1.0"
+
+author:
+  - Shahar Golshani (@shahargolshani)
+
+extends_documentation_fragment:
+  - community.general.attributes
+
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: full
+
+options:
+  path:
+    description:
+      - Path to the directory where files should be removed.
+      - This must be an existing directory.
+    type: path
+    required: true
+
+  pattern:
+    description:
+      - Pattern to match files for removal.
+      - Supports wildcards (V(*), V(?), V([seq]), V([!seq])) for glob-style matching.
+      - Use O(use_regex=true) to interpret this as a regular expression instead.
+    type: str
+    required: true
+
+  use_regex:
+    description:
+      - If V(true), O(pattern) is interpreted as a regular expression.
+      - If V(false), O(pattern) is interpreted as a glob-style wildcard pattern.
+    type: bool
+    default: false
+
+  recursive:
+    description:
+      - If V(true), search for files recursively in subdirectories.
+      - If V(false), only files in the specified directory are removed.
+    type: bool
+    default: false
+
+  file_type:
+    description:
+      - Type of files to remove.
+    type: str
+    choices:
+      file: remove only regular files.
+      link: remove only symbolic links.
+      any: remove both files and symbolic links.
+    default: file
+
+notes:
+  - Directories are never removed by this module, only files and optionally symbolic links.
+  - This module will not follow symbolic links when O(recursive=true).
+  - Be careful with patterns that might match many files, especially with O(recursive=true).
+"""
+
+EXAMPLES = r"""
+- name: Remove all log files from /var/log
+  community.general.file_remove:
+    path: /var/log
+    pattern: "*.log"
+
+- name: Remove all temporary files recursively
+  community.general.file_remove:
+    path: /tmp/myapp
+    pattern: "*.tmp"
+    recursive: true
+
+- name: Remove files matching a regex pattern
+  community.general.file_remove:
+    path: /data/backups
+    pattern: 'backup_[0-9]{8}\.tar\.gz'
+    use_regex: true
+
+- name: Remove both files and symbolic links
+  community.general.file_remove:
+    path: /opt/app/cache
+    pattern: "cache_*"
+    file_type: any
+
+- name: Remove all files starting with 'test_' (check mode)
+  community.general.file_remove:
+    path: /home/user/tests
+    pattern: "test_*"
+  check_mode: true
+"""
+
+RETURN = r"""
+removed_files:
+  description: List of files that were removed.
+  type: list
+  elements: str
+  returned: always
+  sample: ['/var/log/app.log', '/var/log/error.log']
+
+files_count:
+  description: Number of files removed.
+  type: int
+  returned: always
+  sample: 2
+
+path:
+  description: The directory path that was searched.
+  type: str
+  returned: always
```

Comment:
This return value is only returned when `exit_json()` is called.
```suggestion
  returned: success
```